addons/survey/tests/test_survey.py

```diff
@@ -296,3 +297,44 @@ def get_question_by_title(survey, title):
         self.assertNotEqual(q_food_not_vegetarian_cloned.triggering_question_id.id, q_is_vegetarian.id)
         self.assertNotEqual(q_food_not_vegetarian_cloned.triggering_answer_id.id,
                             q_is_vegetarian.suggested_answer_ids[1].id)
+
+    def test_get_questions_except_invalid_conditionals(self):
+        """ Tests the method `_get_questions_except_invalid_conditionals`. This method takes a
+            recordset of question.question from a survey.survey and returns a recordset without
+            conditional questions where the condition is not valid. """
+
+        survey_feedback = self.env.ref('survey.survey_feedback')
+        passed_questions = survey_feedback.question_and_page_ids.filtered(
+            lambda question: not question.is_page or not is_html_empty(question.description)).sorted()
```

Comment:
What is with the `description`?

Reply:
This is what is passed to the method in the code
```
result = self.question_and_page_ids.filtered(
    lambda question: not question.is_page or not is_html_empty(question.description))
result = self._get_pages_or_questions_except_invalid_conditionals(result)
```
When pages have a description, they are shown to the user taking the survey.
To reflect this, I've changed the method names to show that the method returns pages as well

Reply:
Good idea. So we also exclude empty-description pages. 

Perhaps a name like `_get_pages_and_questions_to_show` is more representative, with the docstring specifying the different kind of records excluded?

Side note: About the difficult-to-see difference in indentation at the end of your loop. 
Remember that PEP allows to push the prolonged `for`/`if`/`with`/`while` part one level further to ease readability.
```py
if (something_very_long_that_messes_all_your_layout
        or b):
    print("some stuff")
```